src/frontend/src/optimizer/plan_node/plan_base.rs

```diff
@@ -72,6 +76,7 @@ impl PlanBase {
         functional_dependency: FunctionalDependencySet,
     ) -> Self {
         let id = ctx.next_plan_node_id();
+        let watermark_cols = FixedBitSet::with_capacity(schema.len());
```

Comment:
Almost all plan nodes set their watermark_cols to the `schema.len()` which means it is a placeholder right now?

Reply:
1. `watermark_cols` only has meaning for the stream plan node. before refactoring the plan node is over, we can not express that type-safely 
2. all the stream plan nodes should try their best to derive the watermark_cols, but giving an empty set can give a correct but not good enough plan.

Reply:
Oh, it is an empty set. I get it.